### PR TITLE
Blank Map loaded with call to API (Part 2 of #1934)

### PR DESCRIPTION
**Pull request by sagarpreet-chadha**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

Part 2 of #1934 .

* [X] all tests pass -- `rake test:all`
* [X] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [X] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: app/views/map/_leaflet.html.erb [hunk 1]
```diff
@@ -5,10 +5,21 @@
...
 <% if defined? people %><p><i><small>Share your own location on <a href='/profile'>your profile</a>.</small></i></p><% end %>
 <script>
-  var map = L.map('map<%= unique_id %>').setView([<%= lat %>,<%= lon %>], <%= lat.to_s.length.to_i %> + 6);
+  var map = L.map('map<%= unique_id %>').on('load', onMapLoad).setView([<%= lat %>,<%= lon %>], <%= lat.to_s.length.to_i %> + 6);
   L.tileLayer("//a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(map);
+  function onMapLoad(e){
+	 var lat = <%= lat %> ;
+	 var lon = <%= lon %> ;
+	 var s = lat+","+lon ;
+ 	$.getJSON("https:"+"//publiclab.org/api/srch/locations?srchString="+s , function(data){
```
Comment (jywarren) [inline app/views/map/_leaflet.html.erb]: Should this aim at `/` rather than `//publiclab.org/api/...`?

This is looking FANTASTIC

Comment (sagarpreet-chadha) [reply app/views/map/_leaflet.html.erb]: Hi , thank you so much ! I think both ways are correct , what do you think ?

screen shot 2018-01-25 at 10 53 55 pm

Comment (jywarren) [reply app/views/map/_leaflet.html.erb]: ah, i mean that if you run this codebase somewhere besides publiclab.org, we'd want to just aim the request at the root. Make sense?

**Comment by sagarpreet-chadha** [comment]
@jywarren , i was just testing the code above and when i was using more than one map on a page , only markers of bottom page was showing . Changing the name of javascript object (making unique name for each object) resolved the issue .

**Comment by sagarpreet-chadha** [comment]
I am not quite sure about the URL  (currently the code has `localhost:3000` because i was not able to call `publiclab.org/api/srch/locations?srchString=`  due to CORS constraint ) .
 
I thing we can call this URL `"https:"+"//publiclab.org/api/srch/locations?srchString="+s`  from anywhere because this is an **absolute address** . I am not able to understand how calling root will redirect us to the /api.srch... .
  
I would really like to have some help here ! Thanks 😄 !

**Comment by jywarren** [comment]
ah -- just that if we direct the request at a url starting with `/...` it
will resolve to the current host, so -- `https://publiclab.org/...` on
production, but `localhost:3000/api/srch/loca...` on your local machine.
This will pass CORS restrictions. Should be fine!

On Fri, Jan 26, 2018 at 2:41 AM, Sagarpreet Chadha <notifications@github.com

**Comment by sagarpreet-chadha** [comment]
Awesome ! Got it ! Thanks :-)

**Comment by sagarpreet-chadha** [comment]
Hi , i have done the changes here . Thanks 😄 !

File: app/views/map/_leaflet.html.erb [hunk 2]
```diff
@@ -5,10 +5,21 @@
...
-    L.marker([<%= item.lat %>, <%= item.lon %>]).addTo(map).bindPopup("<a href='<%= item.path %>'><%= item.title %></a>");
-  <% end %>
+  var map<%= unique_id %> = L.map('map<%= unique_id %>').on('load', onMapLoad).setView([<%= lat %>,<%= lon %>], <%= lat.to_s.length.to_i %> + 6);
+  L.tileLayer("//a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(map<%= unique_id %>);
+  function onMapLoad(e){
+	 var lat = <%= lat %> ;
+	 var lon = <%= lon %> ;
+	 var s = lat+","+lon ;
+ 	$.getJSON("/api/srch/locations?srchString="+s , function(data){
+	 			if (!!data.items){
```
Comment (jywarren) [inline app/views/map/_leaflet.html.erb]: Oh, this is looking super, but do you think we could do really consistent two-space indentations here so it's cleanly formatted? Sorry to ask, but it does make a big difference over the long run. Thank you!

Comment (sagarpreet-chadha) [reply app/views/map/_leaflet.html.erb]: Yes @jywarren , i agree and understand the importance of indentation . Doing the changes now , thanks 😄 💯 !

**Comment by sagarpreet-chadha** [comment]
Hi , @jywarren ...just to be clear for future commits , we are following 2 tab spacing or 2 spaces ? 
Thanks 👍 😄 !

**Comment by jywarren** [comment]
2-spaces. @ryzokuken @siaw23 is there something @sagarpreet-chadha could do to auto-format this code, or does Rubocop not read JavaScript? 

I can take a formatting crack at this but I prefer this generally follow this style: https://github.com/publiclab/image-sequencer/blob/master/src/ImageSequencer.js

I wonder if there's a JS style guide we could refer to?

**Comment by ryzokuken** [comment]
@jywarren Now that we're using Rubocop for Ruby, maybe we could also use ESLint for our JavaScript as well?

**Comment by ryzokuken** [comment]
Fun Fact: You can run ESLint with the `--fix` option to autofix most (if not all) errors and warnings 😛

**Comment by jywarren** [comment]
Awesome. Yes, i like the ESLint idea!

**Comment by jywarren** [comment]
I also wanted to note that by moving this into JavaScript, we're no longer able to functional test the map. Perhaps we should write a Jasmine test for it? But we'd have to mock the API response.... it sounds hard. 

This could potentially be made into a separate JS library, independently tested, at some point, the way we do with https://github.com/publiclab/leaflet-blurred-location/

**Pull request merged**